Z_PROBE_ENDSTOP

croadfeldt opened the pull request:
Cleaned up PR for Z_PROBE_ENDSTOP.

croadfeldt committed:
Enabled separate Z Probe and Z Axis endstop use at same time.
Typo fixes in comments in existing code.

croadfeldt committed:
Fixed typo in Z Probe and Endstop section.

croadfeldt committed:
Added credit for code.

croadfeldt committed:
Fix declaration of old_z_probe_endstop.

croadfeldt committed:
Added serial message for Z Probe trigger.

croadfeldt committed:
Filling in more places where Z_PROBE_PIN and Z_PROBE_AND_ENDSTOP need to
be.
Added Sanity Check for it.
Added hook so it's enabled.

croadfeldt committed:
Further Sanity Checks for Z_PROBE_AND_ENDSTOP.

croadfeldt committed:
Bug fixes for Z_PROBE_AND_ENDSTOP.
Code cleanup for Z_PROBE_AND_ENDSTOP.
Added Z_PROBE_PIN to pins_RAMPS_13.h

croadfeldt committed:
Config file cleanup for Z_PROBE_AND_ENDSTOP support.

croadfeldt committed:
Added Z_PROBE_PIN to #define for Z_PROBE_AND_ENDSTOP support.

croadfeldt committed:
Fixed typo for Z_PROBE_AND_ENDSTOP support in pins_RAMPS.h.
Confirmed Sanity checks are working.....

croadfeldt committed:
Additional Sanity Checks for Z_PROBE_AND_ENDSTOP

croadfeldt committed:
More additional sanity checks for Z_PROBE_AND_ENDSTOP

croadfeldt committed:
Config file Z_PROBE_AND_ENDSTOP description clean up.

croadfeldt committed:
More config cleanup work for Z_PROBE_AND_ENDSTOP.

croadfeldt committed:
Don't deploy Z Probe on Z homing if Z_PROBE_AND_ENDSTOP is defined.

croadfeldt committed:
Typo fixed in sanity check.

croadfeldt committed:
Don't deploy probe on Z Axis homing if Z_PROBE_AND_ENDSTOP is enabled,
unless Z_SAFE_HOMING is.

croadfeldt committed:
Ensure Z_PROBE_PIN is defined if Z_PROBE_AND_ENDSTOP is.

croadfeldt committed:
Merge remote-tracking branch 'upstream/development' into development

croadfeldt committed:
Merge remote-tracking branch 'MarlinFirmware/Development' into Development

Conflicts:
	Marlin/Marlin_main.cpp
	Marlin/stepper.cpp

croadfeldt committed:
Don't disable Z_RAISE_BEFORE_HOMING when Z_PROBE_AND_ENDSTOP is enabled.

croadfeldt committed:
Revert 06f767d..cba5692

This rolls back to commit 06f767d608120f09bcd0fd0aee582220cd8657d9.

croadfeldt committed:
Merge remote-tracking branch 'MarlinFirmware/Development' into Development

Conflicts:
	Marlin/Configuration.h

croadfeldt committed:
Merge remote-tracking branch 'origin/Development' into Development

croadfeldt committed:
Manually synching back up with MarlinFirmware/Development.

croadfeldt committed:
Use Z_PROBE_ENDSTOP_INVERTING when checking pin status.

croadfeldt committed:
Fixed Z_PROBE_PIN pullup bug.
Documented some additional areas that should be addressed if Z_PROBE is
fully separated from Z_MIN or Z_MAX.
Fixed a documentation error in sanity checks. Servos start at 0 not 1.

croadfeldt committed:
Merge remote-tracking branch 'MarlinFirmware/Development' into Development

croadfeldt committed:
Merge remote-tracking branch 'MarlinFirmware/Development' into Development

Conflicts:
	Marlin/Marlin_main.cpp

croadfeldt committed:
Merge remote-tracking branch 'MarlinFirmware/Development' into Development

croadfeldt committed:
Changed Z_PROBE_AND_ENDSTOP to Z_PROBE_ENDSTOP.
Updated documentation in Configuration.h.
Cleaned up and commented some code relating to Z_PROBE_ENDSTOP.
Separated Z_MIN_ENDSTOP and Z_PROBE_ENDSTOP completely.

croadfeldt committed:
Merge remote-tracking branch 'MarlinFirmware/Development' into Merge_cleanup

Conflicts:
	Marlin/Marlin_main.cpp

croadfeldt committed:
Merge branch 'Merge_cleanup' into Development

croadfeldt committed:
Cleaning up code in prep for merge with upstream.

croadfeldt committed:
Merge branch 'Merge_cleanup' into Development

croadfeldt committed:
Fixed extra #endif

croadfeldt committed:
Merge branch 'Merge_cleanup' into Development

croadfeldt committed:
Fix Z_PROBE_PING not declared, allows code to compile if Z_PROBE_ENDSTOP
is not used.

croadfeldt committed:
Merge branch 'Merge_cleanup' into Development

croadfeldt committed:
Typo fixed...

thinkyhead commented on Marlin/language.h:
```diff
@@ -138,6 +138,7 @@
 #define MSG_Z_MIN                           "z_min: "
 #define MSG_Z_MAX                           "z_max: "
 #define MSG_Z2_MAX                          "z2_max: "
+#define MSG_Z_PROBE							"z_probe: "
```
Oops, spaces here, not tabs.

thinkyhead commented:
Overall, this is great. We're jumping into a bug-fix mode for a while, so this is on hold till the code is more solid. Later, we might think about calling this `Z_PROBE_SWITCH`, because it's not _really_ an end-stop.

croadfeldt committed:
Spaces not tabs in language.h.
Catch unlikely but possible error and head crash when using
Z_PROBE_REPEATABILITY_TEST

croadfeldt committed:
Not doing network admin work, pin not ping... :)

croadfeldt commented:
Rock on, Z_PROBE_SWITCH works for me. I see my tabs returned in the language file. :) Glad to help out. Looking at the offset thing still. I think what I want to do is already in the code, just need to correctly configure it and then update the documentation a bit.

croadfeldt committed:
Merge branch 'Merge_cleanup' into Development

croadfeldt committed:
Merge remote-tracking branch 'MarlinFirmware/Development' into Merge_cleanup

croadfeldt committed:
Merge branch 'Development' into Merge_cleanup

Natealus commented:
@thinkyhead Wouldn't this help for all the fixes you're doing to bed leveling? I've wondered for a few days if the fact that Z_Probe and Z_MIN_ENDSTOP were kind of the same object in the code, that if it might be causing issues?

thinkyhead commented:
@Natealus Well, this will certainly help eliminate that as a contributing factor. It doesn't amount to a drastic alteration, so I will merge this as part of the effort to deal with any probe=endstop issues, namely that when the probe is an endstop the Z axis is reset to the home position when homed.
